docs/storage/request_queue.md

```diff
@@ -1,50 +1,219 @@
+
+You can use the `RequestQueue` class to specify whether your data is [stored locally or in the Apify cloud](https://sdk.apify.com/docs/api/request-queue), enqueue new URLs
+([see example]((https://sdk.apify.com/docs/examples/puppeteer-crawler))),
+and [manage your existing queues](https://sdk.apify.com/docs/api/request-queue#requestqueueaddrequestrequest-options).
+
+Each actor run is associated with the default request queue, which is created for the actor run, when the first request is added to it. Typically, it is used to store URLs to crawl in the specific actor run, however its usage is optional.
+
+You can also create `named queues` which can be shared between actors or between actor runs.
+
+If you are storing your data locally, you can find your request queue at the following location.
+
+    {APIFY_LOCAL_STORAGE_DIR}/request_queues/{QUEUE_ID}/{STATE}/{NUMBER}.json
+
+> The default request queue's ID is `default`. Each request in the queue is stored as a separate JSON file, where {STATE} is either `handled` or `pending`, and {NUMBER} is an integer indicating the request's position in the queue.
+
+To **open a request queue**, use the `Apify.openRequestQueue()` [method](https://sdk.apify.com/docs/api/apify#apifyopenrequestqueuequeueidorname-options).
+
+    // Import the Apify SDK into your project
     const Apify = require('apify');
 
+    // The main function, which performs the actor's job
+    // and terminates the process when it is finished
     Apify.main(async () => {
-        // Open the default request queue associated with the actor run:
+
+        // Open the default request queue associated with
+        // the actor run
         const queue = await Apify.openRequestQueue();
 
-        // Open a request queue with name "my-queue":
+        // Open the 'my-queue' request queue
         const queueWithName = await Apify.openRequestQueue('my-queue');
     });
 
-If queue is opened then you can use it:
+Once a queue is open, you can manage it using the following methods. For a full list of methods, see the Apify SDK's [API reference](https://sdk.apify.com/docs/api/request-queue).
 
-    // Enqueue some requests
+    // Enqueue requests
     await queue.addRequest({ url: 'http://example.com/aaa' });
-    await queue.addRequest({ url: 'http://example.com/bbb' });
-    await queue.addRequest({ url: 'http://example.com/foo/bar' }, { forefront: true });
-
-    // Get requests from queue
+    await queue.addRequest(
+        { url: 'http://example.com/foo/bar' },
+        { forefront: true }
+    );
+    
+    // Get the next request from queue
     const request1 = await queue.fetchNextRequest();
     const request2 = await queue.fetchNextRequest();
-    const request3 = await queue.fetchNextRequest();
 
-    // Mark a request as handled
-    await queue.markRequestHandled(request1);
+    // Get a specific request
+    const request22 = await queue.getRequest('request-22');
```

Comment:
Maybe use some ID-ish string like `shi6Nh3bfs3` instead of `request-22`. For a moment I was confused by this. But maybe it's just me.